libnavigation-core/src/main/java/com/mapbox/navigation/core/preview/RoutesPreview.kt

```diff
@@ -0,0 +1,42 @@
+package com.mapbox.navigation.core.preview
+
+import androidx.annotation.StringDef
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.core.routealternatives.AlternativeRouteMetadata
+
+@ExperimentalPreviewMapboxNavigationAPI
+data class RoutesPreview(
+    val primaryRoute: NavigationRoute,
+    val alternatives: List<AlternativeRoutePreview>,
+    val routesList: List<NavigationRoute>,
+    val originalRoutesList: List<NavigationRoute>
```

Comment:
Am wondering if two lists is really required here. The primary route is observed through `RoutesObserver`. 

```suggestion
    val alternatives: List<AlternativeRoutePreview>,
    val routes: List<NavigationRoute>,
```

And then define an operation that makes the "active guidance" order.

```
preview[0], preview[1], preview[2], preview[3]
```
If selected index = 2, active guidance order will be mapped to [2, 0, 1, 3]. So the primary route is brought to front.

Reply:
When the `RoutesPreview` object is used to start active navigation, we can create a new `RoutesPreview` object where the `selectedIndex = 0`

Reply:
The operation to create a new list with the order is a relatively cheap operation. 

But if we find need to do it on every route progress update, I can understand the need to cache it instead of creating a new list every time. In that case it should be carefully named and contain good comments.

``` kotlin
/**
 * --snip --
 * @param primaryOrderedRoutes the routes ordered with the primaryIndex route brought to front.
 */
class RoutePreview internal constructor(
  val primaryIndex: Int,
  val routes: List<NavigationRoute>,
  val alternativeRouteMetadata: List<AlternativeRouteMetadata>,
  val primaryOrderedRoutes: List<NavigationRoute>
)
```